tools/vendor.sh

```diff
@@ -0,0 +1,69 @@
+set -e
+
+# Vendors go repository build files and outputs a replacement http_archive.
+#
+# usage ./tools/vendor.sh <org/repo> <commit> [custom-import-path] [custom-name]
+# example: ./tools/vendor.sh docker/docker 363e9a88a11be517d9e8c65c998ff56f774eb4dc
+# example: ./tools/vendor_git.sh googleapis/google-api-go-client c0067489eddd6a0c8aad7c6f9ac0ebd946c0f3d8 google.golang.org/api org_golang_google_api
+
+if [[ $# -eq 0 ]]; then
+    echo """
+    usage ./tools/vendor.sh <org/repo> <commit> [custom-import-path] [custom-name]
+    example: ./tools/vendor.sh docker/docker 363e9a88a11be517d9e8c65c998ff56f774eb4dc
+    example: ./tools/vendor_git.sh googleapis/google-api-go-client c0067489eddd6a0c8aad7c6f9ac0ebd946c0f3d8 google.golang.org/api org_golang_google_api
+    """
+    exit
+fi
+
+pwd=$(pwd)
+
+# Make sure directories exist
+mkdir -p buildpatches
+mkdir -p .buildbuddy-vendor-tmp
+cd .buildbuddy-vendor-tmp
+
+reponame=$(basename "$1" .git)
+archive_url="https://github.com/$1/archive/$2.zip"
+
+# Download the archive
+curl -fsSL -o $reponame.zip $archive_url
+
+# Calculate the sha256
+sha256=$(openssl dgst -sha256 $reponame.zip | awk '{print $2}')
+
+# Unzip the archive
+unzip $reponame.zip
+
+# Go into the unzipped directory
+root_dir=$(unzip -Z -1 $reponame.zip | head -1)
+cd $root_dir
+
+# Fallback to github repo as prefix
+github_prefix="github.com/$1"
+custom_prefix=${3-$github_prefix}
+
+# Fallback to github repo as name
+github_name="com_github_${1/\//_}"
```

Comment:
I think gazelle replaces `-` with `_` as well

Reply:
Ah yeah, I hadn't thought about this too much - switched to all non alphanumeric characters.